qiskit/algorithms/optimizers/gsls.py

```diff
@@ -97,6 +97,10 @@ def get_support_level(self) -> Dict[str, int]:
             "initial_point": OptimizerSupportLevel.required,
         }
 
+    @property
+    def settings(self) -> Dict[str, Any]:
+        return self._options.copy()
```

Comment:
Can you also make this return a dictionary with fixed keys instead of using `self._options`?

Reply:
I changed it to allow only the keys in `GSLS._OPTIONS` 👍🏻